### PR TITLE
Jenish/feature/iris-refactor

### PR DESCRIPTION
**Pull request by JenishPatel99**
Refactoring of Iris
Main points:
- Consolidated APIs for both SPI and UART handling modes
- Move driver APIs to `drivers`
- Fixed return values
- rename files/remove unused files
- Syntax changes
- Format include headers
- Basically cleaned up code where it was needed

### CODE REVIEW

File: Core/Inc/drivers/spi/spi_obc.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ *  Created on: Jul 21, 2022
+ *      Author: jenish
+ */
+
+#ifndef INC_DRIVERS_SPI_SPI_OBC_H_
+#define INC_DRIVERS_SPI_SPI_OBC_H_
+
+#include <iris_system.h>
+
+void obc_spi_transmit(uint8_t *tx_data, uint16_t data_length);
```
Comment (LiamDroog) [inline Core/Inc/drivers/spi/spi_obc.h:13]: The HAL functions in these wrappers return a HAL_StatusTypeDef - we should use them for error checkin

File: Core/Src/obc_handler.c [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                                                  IRIS_UPDATE_CURRENT_LIMIT,
+                                                  IRIS_SET_TIME,
+                                                  IRIS_WDT_CHECK};
+
+/*
+ * Refactor Iris
+ * - No processing of command occurs, delegate it to command_handler.h
+ * - spi drivers should be placed in the driver file **
+ * - Only verification and facilitation of commands should happen hear
+ */
```
Comment (LiamDroog) [inline Core/Src/obc_handler.c:36]: remove

File: Core/Src/obc_handler.c [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        if (iris_commands[index] == obc_cmd) {
+            transmit_ack = 1;
+        }
+    }
+
+    if (transmit_ack != 0) {
+        obc_spi_transmit(&ack, 2);
+        return 0;
+    } else {
+        obc_spi_transmit(&nack, 1);
```
Comment (LiamDroog) [inline Core/Src/obc_handler.c:63]: Why are we sending a 2 byte ack and a 1 byte nack?

File: Core/Src/obc_handler.c [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+/**
+ * @brief
+ * 		Handles command from OBC
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
+ */
+int obc_handle_command(uint8_t obc_cmd) {
+    uint8_t tx_data = 0x69;
```
Comment (LiamDroog) [inline Core/Src/obc_handler.c:77]: nice

Comment (LiamDroog) [reply Core/Src/obc_handler.c:77]: can probably get rid of this and replace with ack

File: Core/Inc/drivers/spi/spi_obc.h (hunk 1 above)
Comment (JenishPatel99) [reply Core/Inc/drivers/spi/spi_obc.h:13]: Agreed

File: Core/Src/obc_handler.c (hunk 2 above)
Comment (JenishPatel99) [reply Core/Src/obc_handler.c]: Done

File: Core/Src/obc_handler.c (hunk 4 above)
Comment (JenishPatel99) [reply Core/Src/obc_handler.c]: Done

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**